### ISSUE TITLE
RKE2 cluster installation fails with calico (v3.24.1) CNI for v1.23.12+rke2r1 onwards because of PSP issue for tigera operator 

### ISSUE THREAD

**Issue by prasundk8s** [rancher/rke2#3532]
**Environmental Info:**
RKE2 Version: v1.23.12+rke2r1

Node(s) CPU architecture, OS, and Version: 
OS SLES 15 SP3
OS  Ubuntu 20.04.3 LTS
CPU: x86-64
tested in multiple different setup/environment with same result

Cluster Configuration:
Typical configuration is 3 master and 3 worker nodes. But this can be easily reproduced with single master node configuration also.

**Describe the bug:**

RKE2 Version: v1.23.12+rke2r1 introduce Calico CNI v3.24.1.  Calico CNI 3.24.1 has transitioned to POD Security standards to support from K8S version 1.25.x onwards. It seems to have a code to check if the version does not support PSP . Please  refer to https://github.com/tigera/operator/pull/2035.  But even in v1.23.12+rke2r1, PSP  for tigera operator installation will not be created during installation and tigera operator installation fails. Calico CNI does not get installed and master node remain in NotReady status. If we manually create the PSP for tigera operator installation ("tigera-operator") following earlier version, calico CNI get installed properly and master node becomes ready. The issue is observed when RKE2 cluter is installed with cis-1.6 profile.

**Steps To Reproduce:**
Please include 

cni: calico
profile: "cis-1.6"

in config.yaml.

Please follow standard installation for v1.23.12+rke2r1.

- Installed RKE2: 

**Expected behavior:**
Calico CNI should get installed properly and master node will become ready

**Actual behavior:**

Master node is NotReady after installation.

NAME       STATUS     ROLES                       AGE     VERSION
clxxxx   NotReady   control-plane,etcd,master   9m22s   v1.23.12+rke2r1

$ kubectl get all -n tigera-operator

NAME                              READY   UP-TO-DATE   AVAILABLE   AGE
deployment.apps/tigera-operator   0/1     0            0           10m

NAME                                         DESIRED   CURRENT   READY   AGE
replicaset.apps/tigera-operator-5c764cb975   1         0         0       10m

**Additional context / logs:**

$ kubectl get ev -n tigera-operator

LAST SEEN   TYPE      REASON              OBJECT                                  MESSAGE
17s         Warning   FailedCreate        replicaset/tigera-operator-5c764cb975   Error creating: pods "tigera-operator-5c764cb975-" 
is forbidden: PodSecurityPolicy: unable to admit pod: [spec.securityContext.hostNetwork: Invalid value: true: Host network is not allowed
to be used spec.volumes[0]: Invalid value: "hostPath": hostPath volumes are not allowed to be 
used]
38s         Normal    ScalingReplicaSet   deployment/tigera-operator              Scaled up replica set tigera-operator-5c764cb975 to 1

If PSP is created manually, this will work,

kubectl create -f tigera-operator.yaml

...

apiVersion: policy/v1beta1
kind: PodSecurityPolicy
metadata:
  annotations:
    meta.helm.sh/release-name: rke2-calico
    meta.helm.sh/release-namespace: kube-system
    seccomp.security.alpha.kubernetes.io/allowedProfileNames: '*'
  labels:
    app.kubernetes.io/managed-by: Helm
  name: tigera-operator
spec:
  allowPrivilegeEscalation: false
  fsGroup:
    ranges:
    - max: 65535
      min: 1
    rule: MustRunAs
  hostNetwork: true
  hostPorts:
  - max: 65535
    min: 0
  requiredDropCapabilities:
  - ALL
  runAsUser:
    rule: MustRunAsNonRoot
  seLinux:
    rule: RunAsAny
  supplementalGroups:
    ranges:
    - max: 65535
      min: 1
    rule: MustRunAs
  volumes:
  - hostPath
  - configMap
  - emptyDir
  - projected
  - secret
  - downwardAPI
  - persistentVolumeClaim

**Comment by brandond** [comment]
cc @manuelbuil

**Comment by rbrtbnfgl** [comment]
what you are saying it's right the tigera-operator should create the PSPs when it started. The issue with CIS 1.6  is that it enforce the PSP and the tigera-operator doesn't start and the PSPs are not created.
The PSPs for the tigera-operator were created on the chart but they were removed on 3.24.1 (https://github.com/projectcalico/calico/commit/3facc075c686ec96600355a4ad0f76882a6e0a57).
I am working on a fix to patch the calico chart to use the tigera-operator PSPs when version is <1.25.

**Comment by VestigeJ** [comment]
Reproduced using the commit just prior to the fix COMMIT=6519e34cfbbf812431d8edfd9c422a07d35c30d3

Cluster persists NotReady status
$ kgn
```
NAME               STATUS     ROLES                       AGE     VERSION
ip-1-1-2-7   NotReady   control-plane,etcd,master   4m36s   v1.23.13+rke2r1
```

$ get.figs rke2
```
=========== rke2 config =========== 
write-kubeconfig-mode: 644
debug: true
token: lurkeyturkey
selinux: true
cni: calico
profile: cis-1.6
```

$ kgp -A
```
NAMESPACE     NAME                                                    READY   STATUS      RESTARTS   AGE
kube-system   cloud-controller-manager-ip-1-1-2-7               1/1     Running     0          4m
kube-system   etcd-ip-1-1-2-7                                   1/1     Running     0          3m54s
kube-system   helm-install-rke2-calico-crd-qwjvs                      0/1     Completed   0          4m22s
kube-system   helm-install-rke2-calico-nbbk8                          0/1     Completed   2          4m22s
kube-system   helm-install-rke2-coredns-xkhnp                         0/1     Completed   0          4m22s
kube-system   helm-install-rke2-ingress-nginx-zjdf8                   0/1     Pending     0          4m22s
kube-system   helm-install-rke2-metrics-server-hrjx7                  0/1     Pending     0          4m22s
kube-system   kube-apiserver-ip-1-1-2-7                         1/1     Running     0          4m
kube-system   kube-controller-manager-ip-1-1-2-7                1/1     Running     0          4m2s
kube-system   kube-proxy-ip-1-1-2-7                             1/1     Running     0          3m44s
kube-system   kube-scheduler-ip-1-1-2-7                         1/1     Running     0          3m54s
kube-system   rke2-coredns-rke2-coredns-775c5b4bb4-9z6vk              0/1     Pending     0          3m52s
kube-system   rke2-coredns-rke2-coredns-autoscaler-695fc554c9-xgm7f   0/1     Pending     0          3m52s
```


Validated fixed using COMMIT=14b39664b6a6430edba31932ef1207ca6a6930ad

$ sudo INSTALL_RKE2_COMMIT=$COMMIT INSTALL_RKE2_EXEC=server ./install-rke2.sh

$ set.kubefig rke2
$ kgn
```
NAME               STATUS   ROLES                       AGE     VERSION
ip-1-1-2-7   Ready    control-plane,etcd,master   3m41s   v1.23.13+rke2r1
```

$ kgp -A
```
NAMESPACE         NAME                                                    READY   STATUS      RESTARTS   AGE
calico-system     calico-kube-controllers-65949964cd-z4qqb                1/1     Running     0          2m30s
calico-system     calico-node-jw8fg                                       1/1     Running     0          2m30s
calico-system     calico-typha-5f9d85667f-6gp42                           1/1     Running     0          2m30s
kube-system       cloud-controller-manager-ip-1-1-2-7               1/1     Running     1          3m37s
kube-system       etcd-ip-1-1-2-7                                   1/1     Running     1          3m8s
kube-system       helm-install-rke2-calico-crd-lwqqq                      0/1     Completed   0          3m22s
kube-system       helm-install-rke2-calico-nr5pz                          0/1     Completed   2          3m22s
kube-system       helm-install-rke2-coredns-4xbfj                         0/1     Completed   0          3m22s
kube-system       helm-install-rke2-ingress-nginx-wb47f                   0/1     Completed   0          3m22s
kube-system       helm-install-rke2-metrics-server-xwvqr                  0/1     Completed   0          3m22s
kube-system       kube-apiserver-ip-1-1-2-7                         1/1     Running     1          3m6s
kube-system       kube-controller-manager-ip-1-1-2-7                1/1     Running     1          3m38s
kube-system       kube-proxy-ip-1-1-2-7                             1/1     Running     0          3m20s
kube-system       kube-scheduler-ip-1-1-2-7                         1/1     Running     1          3m38s
kube-system       rke2-coredns-rke2-coredns-775c5b4bb4-czjx4              1/1     Running     0          2m51s
kube-system       rke2-coredns-rke2-coredns-autoscaler-695fc554c9-tqwj2   1/1     Running     0          2m51s
kube-system       rke2-ingress-nginx-controller-n7786                     1/1     Running     0          101s
kube-system       rke2-metrics-server-669b494888-k4s4c                    1/1     Running     0          116s
tigera-operator   tigera-operator-5c764cb975-5tpvt                        1/1     Running     0          2m35s
```
